subprojects/core/src/main/java/org/gradle/internal/operations/notify/BuildOperationNotificationBridge.java

```diff
@@ -42,19 +41,53 @@
 import java.util.concurrent.locks.Lock;
 import java.util.concurrent.locks.ReentrantLock;
 
-public class BuildOperationNotificationBridge implements Stoppable {
+public class BuildOperationNotificationBridge {
```

Comment:
This class is getting really tough to understand, with 2 listener managers and 3 listeners and a "valve". Some class documentation (and maybe some refactoring if you have ideas how to make it clearer) would be much appreciated.

Reply:
Fair enough. I’ll do that as a separate PR.